From 3194b063c1bcd59966992ed47fc6ae2f4b55d371 Mon Sep 17 00:00:00 2001
From: Yong Gan <yong.gan@nxp.com>
Date: Tue, 7 Aug 2018 09:14:38 +0800
Subject: [PATCH 4299/5242] MGS-3527-2 [#imx-885] enable android native fence
 for Weston sync

commit  820041c08a2304116ab000a75e089f5d5019da8f from
https://source.codeaurora.org/external/imx/linux-imx.git

Fixed the build break for android platform.
set ANDROID_NATIVE_FENCE_SYNC to be 2.

Date: Aug 6, 2018
Signed-off-by: Yong Gan<yong.gan@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mxc/gpu-viv/Kbuild                         |    4 ----
 drivers/mxc/gpu-viv/config                         |    1 -
 .../freescale/gc_hal_kernel_platform_imx.config    |    5 +++--
 3 files changed, 3 insertions(+), 7 deletions(-)

diff --git a/drivers/mxc/gpu-viv/Kbuild b/drivers/mxc/gpu-viv/Kbuild
index 481f9a2..653461c 100644
--- a/drivers/mxc/gpu-viv/Kbuild
+++ b/drivers/mxc/gpu-viv/Kbuild
@@ -309,10 +309,6 @@ else
 EXTRA_CFLAGS += -DgcdENABLE_DRM=0
 endif
 
-ifeq ($(ANDROID_NATIVE_FENCE_SYNC), 1)
-EXTRA_CFLAGS += -DgcdANDROID_NATIVE_FENCE_SYNC=1
-endif
-
 EXTRA_CFLAGS += -I$(AQROOT)/hal/kernel/inc
 EXTRA_CFLAGS += -I$(AQROOT)/hal/kernel
 EXTRA_CFLAGS += -I$(AQROOT)/hal/kernel/arch
diff --git a/drivers/mxc/gpu-viv/config b/drivers/mxc/gpu-viv/config
index 6f6c5e4..9873ae2 100644
--- a/drivers/mxc/gpu-viv/config
+++ b/drivers/mxc/gpu-viv/config
@@ -67,7 +67,6 @@ NONPAGED_MEMORY_CACHEABLE         ?= 0
 NONPAGED_MEMORY_BUFFERABLE        ?= 1
 CACHE_FUNCTION_UNIMPLEMENTED      ?= 0
 USE_BANK_ALIGNMENT                ?= 1
-ANDROID_NATIVE_FENCE_SYNC         ?= 1
 BANK_BIT_START                    ?= 13
 BANK_BIT_END                      ?= 15
 BANK_CHANNEL_BIT                  ?= 12
diff --git a/drivers/mxc/gpu-viv/hal/os/linux/kernel/platform/freescale/gc_hal_kernel_platform_imx.config b/drivers/mxc/gpu-viv/hal/os/linux/kernel/platform/freescale/gc_hal_kernel_platform_imx.config
index 396ae1d..ea83bd9 100644
--- a/drivers/mxc/gpu-viv/hal/os/linux/kernel/platform/freescale/gc_hal_kernel_platform_imx.config
+++ b/drivers/mxc/gpu-viv/hal/os/linux/kernel/platform/freescale/gc_hal_kernel_platform_imx.config
@@ -3,13 +3,14 @@ EXTRA_CFLAGS += -DgcdFSL_CONTIGUOUS_SIZE=134217728
 
 ifneq ($(CONFIG_ANDROID),)
 # build for android
-EXTRA_CFLAGS += -DgcdANDROID_NATIVE_FENCE_SYNC=2
 EXTRA_CFLAGS += -DgcdANDROID
+endif
+
+EXTRA_CFLAGS += -DgcdANDROID_NATIVE_FENCE_SYNC=2
 ifeq ($(CONFIG_SYNC)$(CONFIG_SYNC_FILE),)
 $(warn CONFIG_SYNC or CONFIG_SYNC_FILE is not set in kernel config)
 $(warn Android native fence sync requires CONFIG_SYNC or CONFIG_SYNC_FILE)
 endif
-endif
 
 ifneq ($(CONFIG_ARCH_FSL_IMX8QM),)
 EXTRA_CFLAGS += -DIMX8_SCU_CONTROL=1 -DIMX8_PHYS_BASE=0x80000000 -DIMX8_PHYS_SIZE=0x80000000
-- 
1.7.9.5

